base/compiler/ssair/show.jl

```diff
@@ -372,66 +403,187 @@ function show_ir(io::IO, code::IRCode, expr_type_printer=default_expr_type_print
+        bb_type = length(cfg.blocks[bb_idx].preds) <= 1 ? "─" : "┄"
+        bb_start_str = string(bb_idx_str, " ", bb_type, "─"^bb_pad, " ")
+        bb_guard_rail_cont = string("│  ", " "^max_bb_idx_size)
+        if idx == first(bbrange)
+            bb_guard_rail = bb_start_str
+        else
+            bb_guard_rail = bb_guard_rail_cont
         end
-        if should_print_ssa_type(stmt) && idx in used
-            expr_type_printer(io, typ)
+        # Print linetable information
+        if verbose_linetable
+            stack = compute_loc_stack(code.linetable, code.codelocs[idx])
+            # We need to print any stack frames that did not exist in the last stack
+            ndepth = max(1, length(stack))
+            rail = string(" "^(max_depth+1-ndepth), "│"^ndepth)
+            start_column = cols - max_depth - 10
+            for (i, x) in enumerate(stack)
+                if i > length(last_stack) || last_stack[i] != x
+                    entry = code.linetable[x]
+                    printstyled(io, "\e[$(start_column)G$(rail)\e[1G", color = :light_black)
+                    print(io, bb_guard_rail)
+                    ssa_guard = " "^(maxlength_idx + 4 + (i - 1))
+                    entry_label = "$(ssa_guard)$(entry.method) at $(entry.file):$(entry.line) "
+                    hline = string("─"^(start_column-length(entry_label)-length(bb_guard_rail)+max_depth-i), "┐")
+                    printstyled(io, string(entry_label, hline), "\n"; color=:light_black)
+                    bb_guard_rail = bb_guard_rail_cont
+                end
+            end
+            printstyled(io, "\e[$(start_column)G$(rail)\e[1G", color = :light_black)
+            last_stack = stack
+        else
+            annotation = loc_annotations[idx]
+            loc_method = loc_methods[idx]
+            lineno = loc_lineno[idx]
+            # Print location information right aligned. If the line below is too long, it'll overwrite this,
+            # but that's what we want.
+            if get(io, :color, false)
+                method_start_column = cols - max_method_width - max_loc_width - 2
+                filler = " "^(max_loc_width-length(annotation))
+                printstyled(io, "\e[$(method_start_column)G$(annotation)$(filler)$(loc_method)\e[1G", color = :light_black)
+            end
+            printstyled(io, lineno, " "^(max_lineno_width-length(lineno)+1); color = :light_black)
+        end
+        idx != last(bbrange) && print(io, bb_guard_rail)
+        if idx == last(bbrange) # print separator
+            if idx == first(bbrange)
+                print(io, bb_start_str)
+            elseif idx == last(bbrange)
+                print(io, "└", "─"^(1 + max_bb_idx_size), " ")
+            else
+                print(io, "│  ", " "^max_bb_idx_size)
+            end
+        end
+        if idx == last(bbrange)
+            bb_idx += 1
+        end
+        # convert statement index to labels, as expected by print_stmt
```

Comment:
I'm happier with this now, but now I don't understand it anymore. Doesn't this essentially just duplicate the work of converting to IRCode?